server/asset/eth/rpcclient.go

```diff
@@ -97,74 +104,122 @@ func (c *rpcclient) connect(ctx context.Context, log dex.Logger) error {
 		return fmt.Errorf("no contract address for eth version %d on %s", ethContractVersion, c.net)
 	}
 
-	es, err := swapv0.NewETHSwap(contractAddr, c.ec)
-	if err != nil {
-		return fmt.Errorf("unable to find swap contract: %v", err)
+	c.clients = make([]*ethConn, len(c.endpoints))
+	for i, endpoint := range c.endpoints {
+		client, err := rpc.Dial(endpoint)
```

Comment:
Would `DialContext` not be preferable, so caller can cancel this instead of hang?